src/repr/src/bytes.rs

```diff
@@ -0,0 +1,243 @@
+        let unit_suffix = 'B';
+
+        if bytes < unit {
+            format!("{}B", bytes)
+        } else {
+            let size: f64 = f64::cast_lossy(bytes);
+            let exp: usize = match usize::cast_lossy(size.ln() / unit_base) {
+                e if e == 0 => 1,
+                e => e,
+            };
+
+            format!(
+                "{:.1}{}{}",
+                size / f64::cast_lossy(unit.pow(exp.try_into().expect("usize into u32"))),
+                unit_prefix[exp - 1],
+                unit_suffix
+            )
+        }
+    }
+}
+
+impl Display for Bytes {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.pad(&Self::to_string(self.0))
+    }
+}
+
+impl FromStr for Bytes {
+    type Err = String;
+
+    /// Copied logic from [`bytesize::ByteSize`]
+    fn from_str(value: &str) -> Result<Self, Self::Err> {
+        if let Ok(v) = value.parse::<u64>() {
+            return Ok(Self(v));
+        }
+        let number: String = value
+            .chars()
+            .take_while(|c| c.is_digit(10) || c == &'.')
+            .collect();
+        match number.parse::<f64>() {
+            Ok(v) => {
+                let suffix: String = value
+                    .chars()
+                    .skip_while(|c| c.is_whitespace() || c.is_digit(10) || c == &'.')
+                    .collect();
+                match suffix.parse::<BytesUnit>() {
+                    Ok(u) => Ok(Self(u64::cast_lossy(v * f64::cast_lossy(u.value())))),
+                    Err(error) => Err(format!(
+                        "couldn't parse {:?} into a known SI unit, {}",
+                        suffix, error
+                    )),
+                }
+            }
+            Err(error) => Err(format!(
+                "couldn't parse {:?} into a BytesValue, {}",
+                value, error
+            )),
+        }
+    }
+}
+
+/// Valid units for representing bytes
+#[derive(Debug)]
+pub enum BytesUnit {
+    B,
+    Kb,
+    Mb,
+    Gb,
+    Tb,
+    Pb,
+}
+
+impl BytesUnit {
+    const fn value(&self) -> u64 {
+        match &self {
+            BytesUnit::B => 1,
+            BytesUnit::Kb => 1_024,
+            BytesUnit::Mb => 1_048_576,
+            BytesUnit::Gb => 1_073_741_824,
+            BytesUnit::Tb => 1_099_511_627_776,
+            BytesUnit::Pb => 1_125_899_906_842_624,
+        }
+    }
+}
+
+impl fmt::Display for BytesUnit {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str(match self {
+            BytesUnit::B => "B",
+            BytesUnit::Kb => "KB",
+            BytesUnit::Mb => "MB",
+            BytesUnit::Gb => "GB",
+            BytesUnit::Tb => "TB",
+            BytesUnit::Pb => "PB",
+        })
+    }
+}
+
+impl FromStr for BytesUnit {
+    type Err = String;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        match s.to_uppercase().as_ref() {
+            "B" => Ok(Self::B),
+            "KB" => Ok(Self::Kb),
```

Comment:
```suggestion
            "kB" => Ok(Self::Kb),
```

Reply:
Here it's comparing after `to_uppercase`, but fixed the to_string at other places.